src/node/internal/public_process.ts

```diff
@@ -0,0 +1,408 @@
+  noDeprecation = undefined,
+  permission = undefined,
+  release = undefined,
+  report = undefined,
+  resourceUsage = undefined,
+  send = undefined,
+  traceDeprecation = undefined,
+  throwDeprecation = undefined,
+  sourceMapsEnabled = undefined,
+  threadCpuUsage = undefined,
+  kill = undefined,
+  ref = undefined,
+  unref = undefined;
+
+export function getBuiltinModule(id: string): object {
+  return processImpl.getBuiltinModule(id);
+}
+
+export function exit(code: number): void {
+  processImpl.exitImpl(code);
+}
+
+export function abort(): void {
+  exit(1);
+}
+
+// In future we may return accurate uptime information here, but returning
+// zero at least allows code paths to work correctly in most cases, and is
+// not a completely unreasonable interpretation of Cloudflare's execution
+// model assumptions.
+export function uptime(): number {
+  return 0;
+}
+
+// TODO(soon): Support with proper process.cwd() resolution along with
+//             test in process-nodejs-test.
+export function loadEnvFile(
+  path: string | URL | Buffer = '/bundle/.env'
+): void {
+  if (!compatibilityFlags.experimental || !compatibilityFlags.nodejs_compat) {
+    throw new ERR_UNSUPPORTED_OPERATION();
+  }
+  const { readFileSync } = process.getBuiltinModule('fs') as typeof NodeFS;
+  const parsed = parseEnv(
+    readFileSync(path instanceof URL ? path : path.toString(), 'utf8')
+  );
+  Object.assign(process.env, parsed);
+}
+
+// On the virtual filesystem, we only support group 0
+export function getegid(): number {
+  return 0;
+}
+
+export function getgid(): number {
+  return 0;
+}
+
+export function getgroups(): number[] {
+  return [0];
+}
+
+// On the virtual filesystem, we only support user 0
+export function geteuid(): number {
+  return 0;
+}
+
+export function getuid(): number {
+  return 0;
+}
+
+// We support group 0 but no others, with the virtual filesystem.
+export function setegid(id: number | string): void {
+  if (id !== 0) throw new ERR_EPERM({ syscall: 'setegid' });
+}
+
+// We support group 0 but no others, with the virtual filesystem.
+export function setgid(id: number | string): void {
+  if (id !== 0) throw new ERR_EPERM({ syscall: 'setgid' });
+}
+
+// We support group 0 but no others, with the virtual filesystem.
+export function seteuid(id: number | string): void {
+  if (id !== 0) throw new ERR_EPERM({ syscall: 'seteuid' });
+}
+
+// We support group 0 but no others, with the virtual filesystem.
+export function setuid(id: number | string): void {
+  if (id !== 0) throw new ERR_EPERM({ syscall: 'setuid' });
+}
+
+// We don't support setting or initializing arbitrary groups on the virtual filesystem.
+export function setgroups(_groups: number[]): void {
+  throw new ERR_EPERM({ syscall: 'setgroups' });
+}
+
+export function initgroups(
+  _user: number | string,
+  _extractGroup: number | string
+): void {
+  throw new ERR_EPERM({ syscall: 'initgroups' });
+}
+
+interface Process extends EventEmitter {
```

Comment:
Can you use the type from types/node package rather than defining our own?

Reply:
If I do that, I get this error:

```
Argument of type 'Process' is not assignable to parameter of type 'EventEmitter'.
  Property '[kRejection]' is missing in type 'Process' but required in type 'EventEmitter'.ts(2345)
events.ts(87, 3): '[kRejection]' is declared here.
```

because our event emitter has a non Node.js property typed.